esphome/components/shelly_dimmer/shelly_dimmer.cpp

```diff
@@ -86,20 +93,24 @@ void ShellyDimmer::setup() {
 
       // Firmware upgrade completed, do the checks again.
       continue;
+#else
+      ESP_LOGW(TAG, "Firmware version mismatch, put 'update: True' in the yaml to flash an update.");
```

Comment:
This message might also be good to put in `dump_config` as most people would miss the setup logs if they are connecting wirelessly.
```suggestion
      ESP_LOGW(TAG, "Firmware version mismatch, put 'update: true' in the yaml to flash an update.");
```

Reply:
Ok, I added a dump_config method.